database/database.go

```diff
@@ -87,13 +95,19 @@ func NewCustomDB(ctx context.Context, uri string, dbName string, creds options.C
 	if err != nil {
 		return nil, errors.AddContext(err, "failed to create a new db client")
 	}
-	err = c.Connect(ctx)
+	err = c.Connect(dbCtx)
 	if err != nil {
 		return nil, errors.AddContext(err, "failed to connect to db")
 	}
 	db := c.Database(dbName)
-	err = ensureDBSchema(ctx, db, logger)
-	if err != nil {
+	err = ensureDBSchema(dbCtx, db, logger)
+	if err != nil && errors.Contains(err, ErrIndexCreateFailed) {
+		// We do not error out if we failed to ensure the existence of an index.
+		// It is definitely an issue that should be looked into, which is why we
+		// tag it as [CRITICAL], but seeing as the blocker will work the same
```

Comment:
not entirely correct since there's `unique` indices 🤔  didn't think of that when writing this